pkg/cdi/registry.go

```diff
@@ -90,11 +91,15 @@ type RegistryDeviceDB interface {
 //
 // GetSpecErrors returns any errors for the Spec encountered during
 // the last cache refresh.
+//
+// WriteSpec writes the Spec with the given content and name to the
+// last Spec directory.
 type RegistrySpecDB interface {
 	ListVendors() []string
 	ListClasses() []string
 	GetVendorSpecs(vendor string) []*Spec
 	GetSpecErrors(*Spec) []error
+	WriteSpec(raw *cdi.Spec, name string) error
```

Comment:
Question: Should the writing of the spec be part of a different interface?

Reply:
I deliberately tried to avoid introducing a new one, because TBH I'm not a keen fan of the existing ones either. But if folks consider it better to add a one for this, then please let me know and I'll update the PR accordingly. @bart0sh , @kad ?

Reply:
@klihub can you elaborate a little bit on this? What should and what shouldn't be in this interface in your opinion and why?

Reply:
@bart0sh Sorry, I was too vague in my comment. My comment/dislike was not about the content of the current registry-specific interfaces. It was about the mere existence of those interfaces.

In my original implementation there were no registry-specific extra interfaces, as IIRC there wasn't even a separate `Registry` type at all. The registry was simply a singleton/default instance of the cache and it's full interface was exposed/exported as identically named package-level functions, AFAICT a fairly common golang idiom, a bit similar to how for instance `flag.{Foo(), Xyzzy() and Bar()}` simply alias `flag.CommandLine.{Foo(), Xyzzy(), and Bar()}`.

There was a bit of back and forth and eventually a request to split/introduce the `Registry` interfaces we now have. I agreed to do it so that we could proceed, rework the API to a more usable/consistent set, and get CDI support merged to CRI-O and containerd without much further detail. But, being my admittedly stubborn self, I disliked the idea of the registry being something else than the default instance of the cache already then, and I still dislike it today. Mostly because I don't see any benefit of the extra interfaces, them only causing IMO exactly this kind of unnecessary pain when we extend functionality: which interface should be used in `Registry` to host the new functionality. That's why I tried to avoid introducing one more... 

Does this explain/clarify my earlier comment ?

Reply:
Yes, it does. I tend to agree with you here. I prefer simplicity over complexity, so if not having additional interfaces  simplifies user experience I'm for it. @elezar wdyt?

Reply:
I am happy to remove the interfaces if they're not adding value. It seems as if these were added for a generalisation that was never realised and since the Cache is the only implementation of a registry that we care about (at least at this stage) removing this layer of indirection should not be too problematic.

As a matter of interest, how would returning a cache directly affect our current clients? (`podman` only calls `InjectDevices`).

Reply:
The required changes would probably be very minor. My suggestion would be that if we consider removing the extra interfaces, let's cram the Spec Write interface to the currently suggested interface in this PR. Then take a separate look at removing the interfaces, what kind of exact changes it would require in the affected downstream components then, if we decide to go ahead with it, do that in a separate PR.

Reply:
That sounds good. Ignore my question of whether this is the right interface for this then and consider this thread resolved.

Reply:
Ok, thanks !